ui/ui-components/components/appSelectorList.tsx

```diff
@@ -1,6 +1,34 @@
 import Selector, { BadgeProp } from "./selector";
 import { CardDeck } from "react-bootstrap";
 import { useRouter } from "next/router";
+import { Actions } from "../utils/apiData";
+
+type SelectableApp = Actions.AppOptions["types"][number];
+type SelectablePackage = Actions.PluginsAvailableList["plugins"][number];
+type SelectableSource =
+  Actions.SourceConnectionApps["connectionApps"][number]["app"];
+type SelectableDestination =
+  Actions.DestinationConnectionApps["connectionApps"][number]["app"];
+
+function isSelectableApp(
+  item:
+    | SelectableApp
+    | SelectablePackage
+    | SelectableSource
+    | SelectableDestination
+): item is SelectableApp {
+  return (item as SelectableApp).plugin !== undefined;
+}
```

Comment:
It's probably safe to do:

```suggestion
function isSelectableApp(
  item: any
): item is SelectableApp {
  return item.plugin !== undefined;
}
```